buildlib/tools/git_ready.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# See file LICENSE for terms.
+#
+
+source ./buildlib/tools/codestyle.sh
+
+git_commit() {
+    if ! git diff-index --quiet HEAD; then
+        git add -p
+        # returns 1 if cached index is not empty
+        git diff --cached --exit-code || git commit -m "$title"
+        git reset --hard
```

Comment:
why can't we just leave what remains after `git add -p`? Doing reset looks a bit dangerous to me

Reply:
I think it is ok because we only run this script if tree is completely clean (line 22 early bail otherwise) and `git reset` does not touch untracked files. Also the we do sequence:
1- generate diff (format) + git add -p, commit
2- generate diff (spellcheck) + git add -p, commit
So between the two we need the tree to be clean of leftover changes.